File: fullstacks/rails/app/controllers/api/v1/ai_controller.rb
module Api
  module V1
    class AiController < BaseController
      def suggestion
        render_ok(CapsuleSuggestionService.suggest(title: params[:title], locale: params[:locale]))
      end

      def recommendations
        raw = request.query_parameters["count"]
        count =
          if raw.nil? || raw.to_s.empty?
            4
          else
            n = Integer(raw, exception: false)
            raise ApiError.validation("count 必须是 [3, 8] 范围内的整数", "count") if n.nil? || n < 3 || n > 8

            n
          end
        locale = request.query_parameters["locale"] || "zh-CN"
        render_ok(CapsuleRecommendationService.get_recommendations(count, locale))
      end
    end
  end
end

File: backends/aspnet/src/Controllers/AiController.cs
using HelloTimePro.Aspnet.Dto;
using HelloTimePro.Aspnet.Services;
using HelloTimePro.Aspnet.Web;
using Microsoft.AspNetCore.Mvc;

namespace HelloTimePro.Aspnet.Controllers;

[Route("api/v1")]
public sealed class AiController : ApiControllerBase
{
    private readonly CapsuleSuggestionService _suggestion;
    private readonly CapsuleRecommendationService _recommendation;

    public AiController(CapsuleSuggestionService suggestion, CapsuleRecommendationService recommendation)
    {
        _suggestion = suggestion;
        _recommendation = recommendation;
    }

    [HttpPost("capsule-suggestion")]
    public async Task<IActionResult> Suggest([FromBody] CapsuleSuggestionRequest? req)
    {
        var result = await _suggestion.Suggest(req ?? new CapsuleSuggestionRequest(null, null));
        return Wrapped(result);
    }

    [HttpGet("capsule-recommendations")]
    public async Task<IActionResult> Recommendations()
    {
        var countRaw = StringParam("count");
        int count;
        if (countRaw is not null)
        {
            if (!int.TryParse(countRaw, out var c) || c < 3 || c > 8)
                throw ApiException.Validation("count 必须是 [3, 8] 范围内的整数", "count");
            count = c;
        }
        else
        {
            count = 4;
        }
        var locale = StringParam("locale") ?? "zh-CN";
        return Wrapped(await _recommendation.GetRecommendations(count, locale));
    }
}
